metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart

```diff
@@ -398,10 +399,66 @@ void main() {
       },
     );
 
+    test(
+      ".fetchBuild() throws an ArgumentError if the given pipeline slug is null",
+      () async {
+        expect(() => client.fetchBuild(null, buildNumber), throwsArgumentError);
+      },
+    );
+
+    test(
+      ".fetchBuild() throws an ArgumentError if the given build number is null",
+      () async {
+        expect(
+          () => client.fetchBuild(pipelineSlug, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".fetchBuild() returns an error if the pipeline with the given slug is not found",
+      () async {
+        final interactionResult = await client.fetchBuild(
+          notFound,
+          buildNumber,
+        );
+
+        expect(interactionResult.isError, isTrue);
+      },
+    );
+
+    test(
+      ".fetchBuild() returns an error if the build with the build with the given build number is not found",
+      () async {
+        final interactionResult = await client.fetchBuild(
+          pipelineSlug,
+          notFoundBuildNumber,
+        );
+
+        expect(interactionResult.isError, isTrue);
+      },
+    );
+
+    test(
+      ".fetchBuild() returns a buildkite build",
+      () async {
+        final interactionResult = await client.fetchBuild(
+          pipelineSlug,
+          buildNumber,
+        );
+
+        expect(interactionResult.result, isNotNull);
```

Comment:
Let's also check that the interaction is success here to ensure an algorithm is correct: 
```suggestion
        expect(interactionResult.isSuccess, isTrue);
        expect(interactionResult.result, isNotNull);
```

Should we also check that result has the number as we requested?